api/v1/porterconfig_types_test.go

```diff
@@ -40,33 +39,118 @@ func TestPorterConfigSpec_MergeConfig(t *testing.T) {
-			}},
+	tests := []struct {
+		name        string
+		cfg         PorterConfigSpec
+		expDocument []byte
+	}{
+		{
+			name: "All fields set",
+			cfg: PorterConfigSpec{
+				Debug:                pointer.BoolPtr(true),
+				DebugPlugins:         pointer.BoolPtr(true),
+				Namespace:            pointer.StringPtr("test"),
+				Experimental:         []string{"build-drivers"},
+				BuildDriver:          pointer.StringPtr("buildkit"),
+				DefaultStorage:       pointer.StringPtr("in-cluster-mongodb"),
+				DefaultSecrets:       pointer.StringPtr("keyvault"),
+				DefaultStoragePlugin: pointer.StringPtr("mongodb"),
+				DefaultSecretsPlugin: pointer.StringPtr("kubernetes.secrets"),
+				Storage: []StorageConfig{
+					{PluginConfig{
+						Name:         "in-cluster-mongodb",
+						PluginSubKey: "mongodb",
+						Config:       runtime.RawExtension{Raw: []byte(`{"url":"mongodb://..."}`)},
+					}},
+				},
+				Secrets: []SecretsConfig{
+					{PluginConfig{
+						Name:         "keyvault",
+						PluginSubKey: "azure.keyvault",
+						Config:       runtime.RawExtension{Raw: []byte(`{"vault": "mysecrets"}`)},
+					}},
+				},
+			},
+			expDocument: []byte(`debug: true
+debug-plugins: true
+namespace: test
+experimental:
+    - build-drivers
+build-driver: buildkit
+default-storage: in-cluster-mongodb
+default-secrets: keyvault
+default-storage-plugin: mongodb
+default-secrets-plugin: kubernetes.secrets
+storage:
+    - config:
+        url: mongodb://...
+      name: in-cluster-mongodb
+      plugin: mongodb
+secrets:
+    - config:
+        vault: mysecrets
+      name: keyvault
+      plugin: azure.keyvault
+`),
 		},
-		Secrets: []SecretsConfig{
-			{PluginConfig{
-				Name:         "keyvault",
-				PluginSubKey: "azure.keyvault",
-				Config:       runtime.RawExtension{Raw: []byte(`{"vault": "mysecrets"}`)},
-			}},
+		{
+			name: "Minimum fields set",
+			cfg: PorterConfigSpec{
+				DefaultStoragePlugin: pointer.StringPtr("mongodb"),
+				DefaultSecretsPlugin: pointer.StringPtr("kubernetes.secrets"),
+			},
+			expDocument: []byte(`default-storage-plugin: mongodb
+default-secrets-plugin: kubernetes.secrets
+`),
+		},
+		{
+			name: "Storage Config minimum fields set",
+			cfg: PorterConfigSpec{
+				DefaultStoragePlugin: pointer.StringPtr("mongodb"),
+				DefaultSecretsPlugin: pointer.StringPtr("kubernetes.secrets"),
+				DefaultStorage:       pointer.StringPtr("in-cluster-mongodb"),
+				Storage: []StorageConfig{
+					{PluginConfig{
+						Name:         "in-cluster-mongodb",
+						PluginSubKey: "mongodb",
+					}},
+				},
+			},
+			expDocument: []byte(`default-storage: in-cluster-mongodb 
+default-storage-plugin: mongodb
+default-secrets-plugin: kubernetes.secrets
+storage:
+    name: in-cluster-mongodb
+		plugin: mongodb
+`),
+		},
+		{
+			name: "Secrets Config minimum fields set",
+			cfg: PorterConfigSpec{
+				DefaultStoragePlugin: pointer.StringPtr("mongodb"),
+				DefaultSecretsPlugin: pointer.StringPtr("kubernetes.secrets"),
```

Comment:
Same for this test case, I recommend leaving off DefaultSecretsPlugin.

Reply:
I'll make this a valid config as well